client/repoutils.c

```diff
@@ -276,6 +281,156 @@ TDNFRepoRemoveCache(
+
+    if(!pTdnf || !pTdnf->pConf || IsNullOrEmptyString(pszRepoId))
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    dwError = TDNFRepoGetRpmCacheDir(pTdnf, pszRepoId, &pszRpmCacheDir);
+    BAIL_ON_TDNF_ERROR(dwError);
+
+    if (!IsNullOrEmptyString(pszRpmCacheDir))
+    {
+        dwError = TDNFAllocateStringPrintf(
+                      &pszRpmCacheArchDir,
+                      "%s/%s",
+                      pszRpmCacheDir,
+                      pTdnf->pConf->pszVarBaseArch);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        pDir = opendir(pszRpmCacheArchDir);
+        if(pDir == NULL)
+        {
+           if (errno == ENOENT)
+           {
+               goto cleanup;
+           }
+           dwError = errno;
+           BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+        }
+
+        while ((pEnt = readdir (pDir)) != NULL )
+        {
+             if (!strcmp(pEnt->d_name, ".") || !strcmp(pEnt->d_name, ".."))
+             {
+                 continue;
+             }
+
+             dwError = TDNFAllocateStringPrintf(
+                           &pszFilePath,
+                           "%s/%s",
+                           pszRpmCacheArchDir,
+                           pEnt->d_name);
+             BAIL_ON_TDNF_ERROR(dwError);
+             if(pszFilePath)
+             {
+                if(unlink(pszFilePath))
+                {
+                   dwError = errno;
+                   BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+                }
+                TDNF_SAFE_FREE_MEMORY(pszFilePath);
+                pszFilePath = NULL;
+             }
+             else
+             {
+               dwError = ERROR_TDNF_INVALID_PARAMETER;
+               BAIL_ON_TDNF_ERROR(dwError);
+             }
+        }
+        if(rmdir(pszRpmCacheArchDir))
+        {
+           dwError = errno;
+           BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+        }
+
+        dwError = TDNFAllocateStringPrintf(
+                      &pszRpmCacheNoarchDir,
+                      "%s/%s",
+                      pszRpmCacheDir,
+                      "noarch");
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        pDir = opendir(pszRpmCacheNoarchDir);
+        if(pDir == NULL)
+        {
+           if (errno == ENOENT)
+           {
+               goto cleanup;
+           }
+           dwError = errno;
+           BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+        }
+
+        while ((pEnt = readdir (pDir)) != NULL )
+        {
+             if (!strcmp(pEnt->d_name, ".") || !strcmp(pEnt->d_name, ".."))
+             {
+                 continue;
+             }
+
+             dwError = TDNFAllocateStringPrintf(
+                           &pszFilePath,
+                           "%s/%s",
+                           pszRpmCacheNoarchDir,
+                           pEnt->d_name);
+             BAIL_ON_TDNF_ERROR(dwError);
+             if(pszFilePath)
+             {
+                if(unlink(pszFilePath))
```

Comment:
Is there any chance that pszFilePath could be a directory? if so, this will fail and need manual intervention. Should we just recurse down and delete?

Reply:
No pszFilePath wont be a directory, This is standard rpms directory structure in photon. Even bintray also has the same folder structure. RPMS/x86_64 and RPMS/noarch